api/src/pcapi/core/subscription/educonnect/api.py

```diff
@@ -40,10 +41,13 @@ def handle_educonnect_authentication(
 
     if fraud_check.status == fraud_models.FraudCheckStatus.OK:
         try:
-            subscription_api.activate_beneficiary_if_no_missing_step(user=user)
+            is_activated = subscription_api.activate_beneficiary_if_no_missing_step(user=user)
         except Exception:
             logger.exception("Error while activating user from Educonnect", extra={"user_id": user.id})
             raise exceptions.EduconnectSubscriptionException()
+
+        if not is_activated:
+            users_external.update_external_user(user)
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** Pourquoi ce if ?
Comme je le comprend, on a retiré le `update_external_user` de l'activation, donc pas besoin de vérifier si on y est passé ou pas
‏

Reply:
C'est plutôt : on a retiré `update_external_user ` de la méthode `activate_if_no_missing_step`. 
C'est vraiment une stricte refacto : cette méthode appelait `update_external_user` elle-même si elle n'activait pas l'utilisateur.

Elle ne l'appelle pas si elle l'active car la méthode va être appelée plus tard.
Je suis d'accord, c'est un peu bancal ; idée : avoir un wrapper autour de la route qui l'appelle à la fin, dans tous les cas. Enfin y'a peut-être des fois où on ne veut pas les mettre à jour si on sait qu'il n'y a pas eu de changements... à creuser.